### ISSUE TITLE
mdxComponent works in dev but not on build

### ISSUE THREAD

**Issue by JonathanLorimer** [ElMassimo/iles#245]
I have a bunch of mdx components globally overridden like so:
```typescript
  mdxComponents(){
    return {
      p: Paragraph,
      a: Anchor,
      h1: H1,
      h2: H2,
      ul: UnorderedList,
      ol: OrderedList,
      li: ListItem,
      b: Bold,
      strong: Bold,
      blockquote: Blockquote,
      em: Italic,
      img: Image,
    }
  }
```
and that shows up in dev but not in prod. Note I do have scoped stylesheets in those components, which seem to not be applying. Is that expected?

**Comment by ElMassimo** [comment]
Hi Jonathan!

Can you clarify what you mean with "show up"?

If you provide an example repo, I'd be happy to take a look.

[Here's an example](https://github.com/ElMassimo/iles/blob/c0a3482924afab93081db8dc8174c0aaa6b81b9b/docs/src/app.ts#L27) from the docs.

**Comment by JonathanLorimer** [comment]
Hey! Yes, I will put together a repro repo when I am home. Just was in a rush and wanted to get the ball rolling with the issue!

Thanks for the quick response btw.

**Comment by JonathanLorimer** [comment]
Just for quick reference though, this is what I mean by "not showing up"

```typescript
// app.ts
import Paragraph from "~/components/MDX/Paragraph.vue"
import Anchor from "~/components/MDX/Anchor.vue"
import H1 from "~/components/MDX/H1.vue"
import H2 from "~/components/MDX/H2.vue"
import UnorderedList from "~/components/MDX/UnorderedList.vue"
import ListItem from "~/components/MDX/ListItem.vue"
import OrderedList from "~/components/MDX/OrderedList.vue"
import Bold from "~/components/MDX/Bold.vue"
import Blockquote from "~/components/MDX/Blockquote.vue"
import Italic from "~/components/MDX/Italic.vue"
import Image from "~/components/MDX/Image.vue"

const ph = "phc_gYThVHLoYM3Wuezrv32RQT8ADxySUBGYKAjmk0kb6cv"
const phUrl = "https://app.posthog.com"
const makePosthogScript = (phApiKey: string, phInstanceUrl: string) => `!function(t,e){var o,n,p,r;e.__SV||(window.posthog=e,e._i=[],e.init=function(i,s,a){function g(t,e){var o=e.split(".");2==o.length&&(t=t[o[0]],e=o[1]),t[e]=function(){t.push([e].concat(Array.prototype.slice.call(arguments,0)))}}(p=t.createElement("script")).type="text/javascript",p.async=!0,p.src=s.api_host+"/static/array.js",(r=t.getElementsByTagName("script")[0]).parentNode.insertBefore(p,r);var u=e;for(void 0!==a?u=e[a]=[]:a="posthog",u.people=u.people||[],u.toString=function(t){var e="posthog";return"posthog"!==a&&(e+="."+a),t||(e+=" (stub)"),e},u.people.toString=function(){return u.toString(1)+".people (stub)"},o="capture identify alias people.set people.set_once set_config register register_once unregister opt_out_capturing has_opted_out_capturing opt_in_capturing reset isFeatureEnabled onFeatureFlags".split(" "),n=0;n<o.length;n++)g(u,o[n]);e._i.push([i,s,a])},e.__SV=1)}(document,window.posthog||[]); posthog.init('${phApiKey}', {api_host: '${phInstanceUrl}'})`

export default defineApp({
  mdxComponents: {
    p: Paragraph,
    a: Anchor,
    h1: H1,
    h2: H2,
    ul: UnorderedList,
    ol: OrderedList,
    li: ListItem,
    b: Bold,
    strong: Bold,
    blockquote: Blockquote,
    em: Italic,
    img: Image,
  },
```

```typescript
// ~/components/MDX/Paragraph.vue
<template>
  <p class="mdx-paragraph">
    <slot></slot>
  </p>
</template>

<style>
  .mdx-paragraph {
    font-size: 1.8rem;
    margin-bottom: 4rem;
  }
</style>
```

## Dev
![1680808177_grim](https://user-images.githubusercontent.com/32466011/230472970-e1fbb39b-a6ba-4b94-8bce-de300b7b9651.png)

## Prod
![1680808157_grim](https://user-images.githubusercontent.com/32466011/230473010-8ed08801-1fa7-4d8e-a95d-61217ce708e4.png)

**Comment by JonathanLorimer** [comment]
Here is a minimal repro https://github.com/JonathanLorimer/iles-mdxComponents-repro

**Comment by ElMassimo** [comment]
@JonathanLorimer Thanks for providing details and a reproduction.

Based on what I've found, there's a bug in the MDX optimization for static content, originally added in:

- https://github.com/ElMassimo/iles/pull/66

As a result, the client config with [`overrideElements`](https://github.com/ElMassimo/iles/blob/main/packages/iles/src/node/plugin/plugin.ts#L86) is processed **before** [`detectMdxComponents` is called](https://github.com/ElMassimo/iles/blob/main/packages/iles/src/node/plugin/plugin.ts#L98).

I'll need more time to understand if this bug was present in the first implementation, or is something that started happening on a later Vite version and unfortunately did not manifest in the example apps.

---

This is something that should work out of the box, but as a workaround, you can configure it explicitly for now:

```ts
import { defineConfig } from 'iles'

export default defineConfig({
  markdown: {
    overrideElements: ['p'],
  },
})

```

**Comment by JonathanLorimer** [comment]
@ElMassimo Hey, the fix posted worked. Thanks for the quick responses!

**Comment by ElMassimo** [comment]
Thanks again for reporting, fixed in `iles@0.9.3`.
